src/rtcSessionHelpers.ts

```diff
@@ -74,3 +141,70 @@ export async function leaveRTCSession(
     await widgetPostHangupProcedure(widget);
   }
 }
+
+// Query focus from other room members
+
+// This caches the homeserver domains per homeserver.
+const livekitFocusCache = new Map<string, LivekitFocusConfig | undefined>();
+
+// helper to fetch the domain from a member
+const getDomain = (member: RoomMember): string => member.userId.split(":")[1];
+
+async function fetchAndCacheLivekitFocus(
+  homeserverDomain: string,
+): Promise<LivekitFocusConfig | undefined> {
+  const cachedHomeserverFocus = livekitFocusCache.get(homeserverDomain);
+  if (cachedHomeserverFocus) return cachedHomeserverFocus;
+
+  const fetchedHomeserverFocus = (
+    await AutoDiscovery.getRawClientConfig(homeserverDomain)
+  )[FOCI_WK_KEY] as LivekitFocusConfig | undefined;
```

Comment:
Won't this retrieve an array of focus configs instead of a single one?  If so, perhaps the typechecking+filtering done on the client well-known in `makePreferredFoci` could be split into a function & called here.